### DISCUSSION TITLE
How Bookworm will respond to documents with more complex structures

### DISCUSSION THREAD

**Discussion by cary-rowen** [mush42/bookworm#152] [General]
Hi,

Currently Bookworm uses letters and some other keys to navigate the document, but there are still some special structures that cannot be navigate, such as tables,pictures and math content.

Is it possible for you to implement Ctrl+Alt+Arrow table navigation in Bookworm?

I know Bookworm has a WebViewer, but that feature often requires the user to open the browser and take a long time to find previously read locations, such as pictures and tables

Thanks

**Comment by cary-rowen** [comment]
@mush42

**Comment by mush42** [comment]
I'm a bit busy these days. I'll be back with an answer ASAP.

**Comment by mush42** [comment]
Hello @cary-rowen 

In the beginning, we wanted Bookworm to be a free, high quality alternative to paid eBook readers. I think we have achieved feature parody with those programs.

Most document formats nowadays are based on HTML (or some form of structured markup). The best way to read and interact with such content is via a WebView control (a fully fledged web browser). Unfortunately, using a WebView control will make it impossible to create bookmarks, comments, or highlights. Additionally, it will take much longer to render documents in a WebView instead of a TextControl.
If Bookworm used a WebView control, instead of a TextControl, we would not need any special code to handle navigation and interaction with the content, those would be handled by the screen reader itself.

In summary, we can handle all the types of content if we use a WebView control instead of a TextControl to render content but this will involve some sacrifices.

Please keep in mind that there are HTML only document readers out there. It is important to ask: why people use Bookworm instead of those readers?

Technically speaking, we can imitate all the navigation features that a screen reader provides, but our users need to learn a whole new usage paradigm.

Bookworm strives to be a fully featured, simple, not simplistic, reading solution for blind and visually impaired readers. 

Because this is a discussion, not an issue, I will not give a decisive (yes or no) answer. I'll leave it for the community to take it from here.

Best
Musharraf
